src/rgw/rgw_op.cc

```diff
@@ -3616,15 +3626,15 @@ int RGWPutObj::verify_permission(optional_yield y)
 
     constexpr auto encrypt_attr = "x-amz-server-side-encryption";
     constexpr auto s3_encrypt_attr = "s3:x-amz-server-side-encryption";
-    auto enc_header = s->info.x_meta_map.find(encrypt_attr);
-    if (enc_header != s->info.x_meta_map.end()){
+    auto enc_header = s->info.crypt_attribute_map.find(encrypt_attr);
+    if (enc_header != s->info.crypt_attribute_map.end()){
       rgw_add_to_iam_environment(s->env, s3_encrypt_attr, enc_header->second);
```

Comment:
i'm skeptical that bucket policy should see the transformed sse attributes (based on bucket encryption policy)

and it's not clear whether `get_encryption_defaults()` has performed those transformations yet

Reply:
I believe get_encryption_defaults is called before this.  And it's unclear to me whether bucket policy sohuld see bucket encryption defaults or not.  I'm hopiong Ali will get us an ansewr on that.

Reply:
ok. in response to your email, i argued that the defaults shouldn't be visible to bucket policy. but that was based on an assumption that all request headers were visible to iam like `s3:x-amz-server-side-encryption`, but this function shows that's clearly not the case

i vote that we leave is this way for initial merge, and follow up with testing and docs based on what @alimaredia learns here